src/rbus/rbus.c

```diff
@@ -5356,13 +5678,32 @@ rbusError_t  rbusEvent_PublishRawData(
     {
         return RBUS_ERROR_NOSUBSCRIBERS;
     }
+    rtList_GetFront(el->subscriptions, &listItem);
+    while(listItem)
+    {
+        rtListItem_GetData(listItem, (void**)&subscription);
+        if(!subscription || !subscription->eventName || !subscription->listener)
+        {
+            RBUSLOG_INFO("rbusEvent_Publish failed: null subscriber data");
+            if(rc == RBUS_ERROR_SUCCESS)
+                rc = RBUS_ERROR_BUS_ERROR;
+            rtListItem_GetNext(listItem, &listItem);
+        }
+        const rtPrivateClientInfo *pPrivCliInfo = _rbuscore_find_server_privateconnection (subscription->eventName, subscription->listener);
```

Comment:
i see why you declared this in rbuscore.h but this function already misses the mutex. So, using here without protection is not safe.
So better to have a rbuscore wrapper function which acquires mutex before accessing the pPrivCliInfo and publishes it.